### ISSUE TITLE
[Proposal] Sync method for one-to-many relation

### ISSUE THREAD

**Issue by FinesseRus** [laravel/ideas#1144]
It should behave the same way as the `sync` method of `many-to-many` relation. In should make the `one` side object have only the specified objects on the `many` side:

```php
$comments = getCommentsFromSomewhere();
$post->comments()->sync($comments);
```

Currently it can be replaced with the following code:

```php
$comments = getCommentsFromSomewhere();
$post->comments()->whereNotIn('id', $comments->pluck('id'))->update(['post_id' => null]);
$post->comments()->saveMany($comments);
```

**Comment by AkenRoberts** [comment]
What if someone doesn't want to orphan the data, and rather delete it? Which is `sync()` supposed to do?

**Comment by sisve** [comment]
> Also a model can have more then one parent (e.g. a post can have a creator and a reviewer). In this case living an object without a parent doesn’t make it orphan.

This is what would differ between a has-one relation to a belongs-to relation. But the issue title mentions one-to-many, so this specific scenario wouldn't apply.

**Comment by FinesseRus** [comment]
@sisve

> This is what would differ between a has-one relation to a belongs-to relation. But the issue title mentions one-to-many, so this specific scenario wouldn't apply.

No, it is a belongs-to (and has-many on the opposite side) relation. Considering my example, a user can have many created posts and many reviewed posts, but a post can have only one creator and at most one reviewer.
